services/diagnostic/handlers.go

```diff
@@ -0,0 +1,1222 @@
+	h.mu.Lock()
+	defer h.mu.Unlock()
+	defer h.buf.Reset()
+	fmt.Fprintln(h.buf, ctx...)
+
+	h.l.Debug(strconv.Quote(h.buf.String()))
+}
+
+func (h *ScraperHandler) Debugf(s string, ctx ...interface{}) {
+	h.l.Debug(fmt.Sprintf(s, ctx...))
+}
+
+func (h *ScraperHandler) Info(ctx ...interface{}) {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+	defer h.buf.Reset()
+	fmt.Fprint(h.buf, ctx...)
+
+	h.l.Info(h.buf.String())
+}
+
+func (h *ScraperHandler) Infoln(ctx ...interface{}) {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+	defer h.buf.Reset()
+	fmt.Fprintln(h.buf, ctx...)
+
+	h.l.Info(strconv.Quote(h.buf.String()))
+}
+
+func (h *ScraperHandler) Infof(s string, ctx ...interface{}) {
+	h.l.Debug(fmt.Sprintf(s, ctx...))
+}
+
+func (h *ScraperHandler) Warn(ctx ...interface{}) {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+	defer h.buf.Reset()
+	fmt.Fprint(h.buf, ctx...)
+
+	h.l.Warn(h.buf.String())
+}
+
+func (h *ScraperHandler) Warnln(ctx ...interface{}) {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+	defer h.buf.Reset()
+	fmt.Fprintln(h.buf, ctx...)
+
+	h.l.Warn(strconv.Quote(h.buf.String()))
+}
+
+func (h *ScraperHandler) Warnf(s string, ctx ...interface{}) {
+	h.l.Warn(fmt.Sprintf(s, ctx...))
+}
+
+func (h *ScraperHandler) Error(ctx ...interface{}) {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+	defer h.buf.Reset()
+	fmt.Fprint(h.buf, ctx...)
+
+	h.l.Error(h.buf.String())
+}
+
+func (h *ScraperHandler) Errorln(ctx ...interface{}) {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+	defer h.buf.Reset()
+	fmt.Fprintln(h.buf, ctx...)
+
+	h.l.Error(strconv.Quote(h.buf.String()))
+}
+
+func (h *ScraperHandler) Errorf(s string, ctx ...interface{}) {
+	h.l.Error(fmt.Sprintf(s, ctx...))
+}
+
+func (h *ScraperHandler) Fatal(ctx ...interface{}) {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+	defer h.buf.Reset()
+	fmt.Fprint(h.buf, ctx...)
+
+	h.l.Fatal(h.buf.String())
+}
+
+func (h *ScraperHandler) Fatalln(ctx ...interface{}) {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+	defer h.buf.Reset()
+	fmt.Fprintln(h.buf, ctx...)
+
+	h.l.Fatal(h.buf.String())
+}
+
+func (h *ScraperHandler) Fatalf(s string, ctx ...interface{}) {
+	h.l.Fatal(fmt.Sprintf(s, ctx...))
+}
+
+func (h *ScraperHandler) With(key string, value interface{}) plog.Logger {
+	var field zapcore.Field
+
+	switch value.(type) {
+	case int:
+		field = zap.Int(key, value.(int))
+	case float64:
+		field = zap.Float64(key, value.(float64))
+	case string:
+		field = zap.String(key, value.(string))
+	case time.Duration:
+		field = zap.Duration(key, value.(time.Duration))
+	default:
+		field = zap.String(key, fmt.Sprintf("%v", value))
+	}
+
+	return &ScraperHandler{
+		l: h.l.With(field),
+	}
+}
+
+func (h *ScraperHandler) SetFormat(string) error {
+	return nil
+}
+
+func (h *ScraperHandler) SetLevel(string) error {
+	return nil
+}
+
+// Edge Handler
+
+type EdgeHandler struct {
+	l *zap.Logger
+}
+
+func (h *EdgeHandler) Collect(mtype edge.MessageType) {
+	h.l.Debug("collected message", zap.Stringer("message_type", mtype))
```

Comment:
Does this mean we are logging every message?

Reply:
I think this comes from https://github.com/influxdata/kapacitor/blob/md-log-v2/edge/log.go#L24-L35 and I don't think the `LogEdge` is ever used. At least I don't think that it is.